src/Appwrite/Migration/Migration.php

```diff
@@ -195,17 +195,23 @@ public function forEachDocument(callable $callback): void
      * @return iterable<Document>
      * @throws \Exception
      */
-    public function documentsIterator(string $collectionId): iterable
+    public function documentsIterator(string $collectionId, $queries = []): iterable
     {
         $sum = 0;
         $nextDocument = null;
         $collectionCount = $this->projectDB->count($collectionId);
+        $queries[] = Query::limit($this->limit);
 
         do {
-            $queries = [Query::limit($this->limit)];
             if ($nextDocument !== null) {
-                $queries[] = Query::cursorAfter($nextDocument);
+                $lastQuery = $queries[\count($queries) - 1];
```

Comment:
What if the `cursorAfter` is not the last query passed but the first of multiple?